Subject: [PATCH] Add @type to some definitions
---
Index: src/main/webui/src/generated/proposalToolSchemas.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/webui/src/generated/proposalToolSchemas.ts b/src/main/webui/src/generated/proposalToolSchemas.ts
--- a/src/main/webui/src/generated/proposalToolSchemas.ts	(revision e7cd9701848b1398ff55f5006a6c8729a02fadb7)
+++ b/src/main/webui/src/generated/proposalToolSchemas.ts	(date 1690970029070)
@@ -166,6 +166,7 @@
  * The basic target setup for a celestial Target.
  */
 export type CelestialTarget = {
+  "@type"?: string; // proposal:CelestialTarget
   /**
    * A common name for the source
    */
@@ -494,6 +495,7 @@
 };
 
 export type Investigator = {
+  "@type"?: string;
   type?: InvestigatorKind;
   /**
    * is the investigator making proposal for their PhD
@@ -1344,8 +1346,9 @@
  * A target source
  */
 export type Target = {
+  "@type"?: string;
   /**
-   * A common name for the source
+   * /A common name for the source
    */
   sourceName?: string;
 };
