Subject: [PATCH] Patch to add @type to some definitions
---
Index: src/main/webui/src/generated/proposalToolSchemas.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/webui/src/generated/proposalToolSchemas.ts b/src/main/webui/src/generated/proposalToolSchemas.ts
--- a/src/main/webui/src/generated/proposalToolSchemas.ts	(revision c487967e5e0da659f2c76006549e53ed8671ffe4)
+++ b/src/main/webui/src/generated/proposalToolSchemas.ts	(date 1690982315638)
@@ -166,6 +166,7 @@
  * The basic target setup for a celestial Target.
  */
 export type CelestialTarget = {
+  "@type"?: string; // proposal:CelestialTarget
   /**
    * A common name for the source
    */
@@ -331,6 +332,7 @@
  * A Point on the Unit Sphere
  */
 export type EquatorialPoint = {
+  coordSys: CoordSys;
   /**
    * A real value with a unit.
    */
@@ -645,6 +648,7 @@
  * a complete proposal
  */
 export type ObservingProposal = {
+  "_id"?: number;
   /**
    * the proposal title
    */
@@ -768,6 +772,7 @@
  * Abstract head of coordinate spaces related to physical properties.
  */
 export type PhysicalCoordSpace = {
+  "@type": string;
   /**
    * Describes an axis of the coordinate space.
    */
@@ -1208,6 +1213,7 @@
  * Specialized coordinate system for the Spatial domain. This object SHOULD include an appropriate SpaceFrame. In Appendix B, we define two standard spatial coordinate space instances (Spherical and Cartesian), which may be referenced in serializations. If a CoordSpace is not provided, it is assumed to be represented by a Standard Spherical Coordinate Space.
  */
 export type SpaceSys = {
+  "@type": string;
   /**
    * Abstract head of coordinate spaces related to physical properties.
    */
@@ -1344,6 +1350,7 @@
  * A target source
  */
 export type Target = {
+  "@type"?: string;
   /**
    * A common name for the source
    */