--- proposalToolSchemas.ts.orig	2025-11-13 16:17:25
+++ proposalToolSchemas.ts	2025-12-18 15:08:51
@@ -56,6 +56,8 @@
  * A block of resources that have been allocated
  */
 export type AllocatedBlock = {
+  _id?: number;
+  "@type"?: string; //proposalManagement:AllocatedBlock
   /**
    * a configuration can be used to observe with.
    */
@@ -74,6 +76,7 @@
  * an instance of a proposal that is allocated observing time
  */
 export type AllocatedProposal = {
+  _id?: number;
   /**
    * an instance of a proposal that has been submitted
    */
@@ -88,6 +91,7 @@
  * The final grade given by the TAC
  */
 export type AllocationGrade = {
+  _id?: number;
   /**
    * the name of the grade
    */
@@ -131,6 +135,7 @@
  * a processing backend /pipeline- e.g. correlator
  */
 export type Backend = {
+  _id?: number
   /**
    * the backend name
    */
@@ -179,6 +184,7 @@
  */
 export type CalibrationObservation = {
   xmlId?: string;
+  "@type"?: string;
   /**
    * A reference to - The actual target of the observation
    */
@@ -212,6 +218,7 @@
  * Spatial domain, three-dimensional cartesian coordinate space. The particulars of the axis descriptions depend on the physical constraints of the instance. In Appendix B, we provide the description of a Standard Cartesian Coordinate Space instance which applies to many Astronomical cases, and may be referenced in serializations.
  */
 export type CartesianCoordSpace = {
+  "@type"?: string; // coords:CartesianCoordSpace
   axis?: Axis[];
 };
 
@@ -238,6 +245,7 @@
  */
 export type CelestialTarget = {
   xmlId?: string;
+  "@type"?: string; // proposal:CelestialTarget
   /**
    * A common name for the source
    */
@@ -339,7 +347,9 @@
  * Abstract head of the coordinate system object tree.
  */
 export type CoordSys = {
+  "@type"?: string;
   xmlId?: string;
+  frame?: CoordFrame;
 };
 
 /**
@@ -491,6 +501,8 @@
  * A Point on the Unit Sphere
  */
 export type EquatorialPoint = {
+  "@type": string; // coords:EquatorialPoint
+  coordSys: CoordSys;
   /**
    * A real value with a unit.
    */
@@ -537,8 +549,10 @@
  * Definition of an observing field pointing
  */
 export type Field = {
+  "@type"?: string;
   name?: string;
   xmlId?: string;
+  _id?: number;
 };
 
 export type FileUpload = Record<string, any>;
@@ -547,6 +561,7 @@
  * Available filters /bandpasses for intruments
  */
 export type Filter = {
+  _id?: number
   /**
    * human readable name
    */
@@ -623,6 +638,7 @@
  * An instrument that can be attached to a telescope - e.g. CCD, Radio Receiver
  */
 export type Instrument = {
+  _id?: number
   /**
    * the instrument name - human readable
    */
@@ -672,10 +688,11 @@
 };
 
 export type Investigator = {
-  /**
-   * person connected with the proposal
-   */
-  person?: Person;
+  _id?: number;
+    /**
+     * person connected with the proposal
+     */
+    person?: Person;
   type?: InvestigatorKind;
   /**
    * is the investigator making proposal for their PhD
@@ -767,6 +784,8 @@
  * An observation - a pointing of the telescope at a part of the sky, occurs in a single non-overlapping time period
  */
 export type Observation = {
+  "@type"?: string; //ObsType (see above)
+  _id?: number;
   /**
    * A reference to - The actual target of the observation
    */
@@ -815,6 +834,9 @@
  * An organisation that can perform astronomical observations
  */
 export type Observatory = {
+  "@type"?: string;
+  "_id"?: number;
+
   xmlId?: string;
   /**
    * The name of the organization
@@ -860,6 +882,7 @@
  * a configuration can be used to observe with.
  */
 export type ObservingMode = {
+  _id?: number;
   /**
    * human readable name for the mode
    */
@@ -891,6 +914,7 @@
  * base type of a telescope or array of telescopes
  */
 export type ObservingPlatform = {
+  _id?: number
   xmlId?: string;
 };
 
@@ -898,6 +922,7 @@
  * a complete proposal
  */
 export type ObservingProposal = {
+  _id?: number;
   xmlId?: string;
   /**
    * the proposal title
@@ -947,6 +972,8 @@
  * An institution that is a collection of people
  */
 export type Organization = {
+  "@type"?: string; // proposal:Organization
+  _id?: number;
   /**
    * The name of the organization
    */
@@ -1010,6 +1037,7 @@
    */
   orcidId?: StringIdentifier;
   xmlId?: string;
+  _id?: number;
 };
 
 /**
@@ -1171,10 +1199,12 @@
  * Defines collection of resources and proposals for a particular observing season
  */
 export type ProposalCycle = {
+  _id?: number
   /**
-   * An organisation that can perform astronomical observations
-   */
+  * An organisation that can perform astronomical observations
+  */
   observatory?: Observatory;
+
   /**
    * a human readable description of the cycle
    */
@@ -1249,6 +1279,7 @@
  * A review of a proposal
  */
 export type ProposalReview = {
+  _id?: number;
   /**
    * assigned to review the proposal
    */
@@ -1312,6 +1343,7 @@
  * A real value with a unit.
  */
 export type RealQuantity = {
+  "@type"?: string; // ivoa:RealQuantity
   /**
    * Must conform to definition of unit in VOUnit spec.
    */
@@ -1343,6 +1375,7 @@
  * A resource that will be consumed by allocating an observation from a proposal
  */
 export type Resource = {
+  _id?: number;
   /**
    * a type of resource
    */
@@ -1373,6 +1406,7 @@
  * a type of resource
  */
 export type ResourceType = {
+  _id?: number;
   /**
    * the name of the resource type
    */
@@ -1385,6 +1419,7 @@
  * assigned to review the proposal
  */
 export type Reviewer = {
+  _id?: number;
   /**
    * person connected with the proposal
    */
@@ -1416,6 +1451,8 @@
   /**
    * Science oriented definition of a spectral window.
    */
+  _id?: number;
+
   spectralWindowSetup?: SpectralWindowSetup;
   expectedSpectralLine?: ExpectedSpectralLine[];
 };
@@ -1446,9 +1483,10 @@
  * A SpaceFrame is specified by its reference frame (orientation), and a reference position (origin). Currently only standard reference frames are allowed. An equinox MUST be provided for pre-ICRS reference frames. A planetary ephemeris MAY be provided if relevant. If needed, but not provided, it is assumed to be 'DE405'.
  */
 export type SpaceFrame = {
+  "@type": string; // coords:SpaceFrame
   /**
-   * RefLocation defines the origin of the spatial coordinate space. This location is represented either by a standard reference position (for which the absolute location in phase space is known by definition), or a specified point in another Spatial frame. This object is used as the origin of the SpaceFrame here, but also to specify the Spatial Reference Position (refPosition) associated with other domain Frames. For example, in the Time domain, the Spatial Reference Position indicates that the 'time' values are the time that the 'event' occured at that location, which might be different from the detector location.
-   */
+  * RefLocation defines the origin of the spatial coordinate space. This location is represented either by a standard reference position (for which the absolute location in phase space is known by definition), or a specified point in another Spatial frame. This object is used as the origin of the SpaceFrame here, but also to specify the Spatial Reference Position (refPosition) associated with other domain Frames. For example, in the Time domain, the Spatial Reference Position indicates that the 'time' values are the time that the 'event' occured at that location, which might be different from the detector location.
+  */
   refPosition?: RefLocation;
   /**
    * The spatial reference frame. Values MUST be selected from the controlled vocabulary at the given URL.
@@ -1468,6 +1506,7 @@
  * Specialized coordinate system for the Spatial domain. This object SHOULD include an appropriate SpaceFrame. In Appendix B, we define two standard spatial coordinate space instances (Spherical and Cartesian), which may be referenced in serializations. If a CoordSpace is not provided, it is assumed to be represented by a Standard Spherical Coordinate Space.
  */
 export type SpaceSys = {
+  "@type": string; // coords:SpaceSys
   xmlId?: string;
   /**
    * Abstract head of coordinate spaces related to physical properties.
@@ -1551,7 +1590,7 @@
   /**
    * person connected with the proposal
    */
-  person?: Person | Person;
+  person?: Person;
   uid?: string;
   inKeycloakRealm?: boolean;
 };
@@ -1571,6 +1610,8 @@
  * an instance of a proposal that has been submitted
  */
 export type SubmittedProposal = {
+  _id?: number;
+
   xmlId?: string;
   /**
    * the proposal title
@@ -1696,17 +1737,20 @@
  * A target source
  */
 export type Target = {
+  "@type"?: string;
   /**
    * A common name for the source
    */
   sourceName?: string;
   xmlId?: string;
+  _id?: number;
 };
 
 /**
  * the field points to the associated target
  */
 export type TargetField = {
+  "@type"?: string;
   xmlId?: string;
   name?: string;
 };
@@ -1715,6 +1759,7 @@
  * an observation of the scientific target
  */
 export type TargetObservation = {
+  "@type"?: string;
   xmlId?: string;
   /**
    * A reference to - The actual target of the observation
@@ -1743,10 +1788,13 @@
    */
   performance?: PerformanceParameters;
   spectrum?: ScienceSpectralWindow[];
+  _id?: number;
   xmlId?: string;
 };
 
 export type Telescope = {
+  _id?: number
+  "@type"?: string;
   xmlId?: string;
   /**
    * telescope name
@@ -1766,6 +1814,7 @@
  * a set of telescopes that are operated together for an observation
  */
 export type TelescopeArray = {
+  "@type"?: string;
   xmlId?: string;
   /**
    * the array name
@@ -1833,6 +1882,10 @@
  * particular time range
  */
 export type TimingWindow = {
+  "@type"?: string; //proposal:TimingWindow
+
+  _id?: number;
+
   note?: string;
   isAvoidConstraint?: boolean;
   /**
