--- proposalToolSchemas.ts.orig	2026-05-28 09:34:46
+++ proposalToolSchemas.ts	2026-05-28 10:02:12
@@ -16,11 +16,11 @@
    */
   summary?: string;
   /**
-   * scientific justification
+   * The justification for the proposal. Note that the justification is for reading by humans and is not parsed to extract things like source lists - that sort of information must be entered in the correct place in the model.
    */
   scientificJustification?: Justification;
   /**
-   * technical justification
+   * The justification for the proposal. Note that the justification is for reading by humans and is not parsed to extract things like source lists - that sort of information must be entered in the correct place in the model.
    */
   technicalJustification?: Justification;
   /**
@@ -59,16 +59,18 @@
  * A block of resources that have been allocated
  */
 export type AllocatedBlock = {
+  _id?: number;
+  "@type"?: string; //proposalManagement:AllocatedBlock
   /**
-   * A reference to -
+   * a configuration can be used to observe with.
    */
   mode?: ObservingMode;
   /**
-   * the amount of resource
+   * A resource that will be consumed by allocating an observation from a proposal
    */
   resource?: Resource;
   /**
-   * A reference to -
+   * The final grade given by the TAC
    */
   grade?: AllocationGrade;
 };
@@ -77,8 +79,9 @@
  * an instance of a proposal that is allocated observing time
  */
 export type AllocatedProposal = {
+  _id?: number;
   /**
-   * A reference to - the proposal
+   * an instance of a proposal that has been submitted
    */
   submitted?: SubmittedProposal;
   /**
@@ -91,6 +94,7 @@
  * The final grade given by the TAC
  */
 export type AllocationGrade = {
+  _id?: number;
   /**
    * the name of the grade
    */
@@ -113,6 +117,7 @@
  * a processing backend /pipeline- e.g. correlator
  */
 export type Backend = {
+  _id?: number
   /**
    * the backend name
    */
@@ -143,16 +148,17 @@
  */
 export type CalibrationObservation = {
   xmlId?: string;
+  "@type"?: string;
   /**
    * A reference to - The actual target of the observation
    */
   target?: Target[];
   /**
-   * A reference to - The Field for the observation
+   * Definition of an observing field pointing
    */
   field?: Field;
   /**
-   * A reference to - The technical goals of the observation
+   * collects together the technical goals of the proposal
    */
   technicalGoal?: TechnicalGoal;
   /**
@@ -177,15 +183,15 @@
 
 export type CartesianPoint = {
   /**
-   * x coord
+   * A real value with a unit.
    */
   x?: RealQuantity;
   /**
-   * y coord
+   * A real value with a unit.
    */
   y?: RealQuantity;
   /**
-   * z coord
+   * A real value with a unit.
    */
   z?: RealQuantity;
   /**
@@ -214,12 +220,13 @@
  */
 export type CelestialTarget = {
   xmlId?: string;
+  "@type"?: string; // proposal:CelestialTarget
   /**
    * A common name for the source
    */
   sourceName?: string;
   /**
-   * the coordinates
+   * A Point on the Unit Sphere
    */
   coord?: CelestialPosition;
   /**
@@ -257,9 +264,12 @@
  */
 export type CommitteeMember = {
   /**
-   * A reference to - the member
+   * assigned to review the proposal
    */
   member?: Reviewer;
+  /**
+   * A role within the timeAllocation committee
+   */
   role?: TacRole;
 };
 
@@ -361,7 +371,7 @@
 };
 
 /**
- * an elliptical region on the sky
+ * Specialization of a Field for an elliptical map.
  */
 export type Ellipse = {
   /**
@@ -408,14 +418,17 @@
  * Definition of an observing field pointing
  */
 export type Field = {
+  "@type"?: string;
   name?: string;
   xmlId?: string;
+  _id?: number;
 };
 
 /**
- * Available filters /bandpasses for intruments
+ * Available filters /bandpasses for instruments
  */
 export type Filter = {
+  _id?: number
   /**
    * human readable name
    */
@@ -434,6 +447,7 @@
  * An instrument that can be attached to a telescope - e.g. CCD, Radio Receiver
  */
 export type Instrument = {
+  _id?: number
   /**
    * the instrument name - human readable
    */
@@ -474,7 +488,7 @@
  */
 export type IntegerQuantity = {
   /**
-   * The unit of this quantity.
+   * Must conform to definition of unit in VOUnit spec.
    */
   unit?: Unit;
   /**
@@ -500,8 +514,9 @@
 };
 
 export type Investigator = {
+  _id?: number;
   /**
-   * A reference to - the investigator
+   * person connected with the proposal
    */
   person?: Person;
   /**
@@ -532,7 +547,7 @@
    */
   text?: string;
   /**
-   * the format of the text
+   * acceptable text formats for document submission
    */
   format?: TextFormats;
 };
@@ -587,16 +602,18 @@
  * An observation - a pointing of the telescope at a part of the sky, occurs in a single non-overlapping time period
  */
 export type Observation = {
+  "@type"?: string; //ObsType (see above)
+  _id?: number;
   /**
    * A reference to - The actual target of the observation
    */
   target?: Target[];
   /**
-   * A reference to - The Field for the observation
+   * Definition of an observing field pointing
    */
   field?: Field;
   /**
-   * A reference to - The technical goals of the observation
+   * collects together the technical goals of the proposal
    */
   technicalGoal?: TechnicalGoal;
   /**
@@ -626,7 +643,7 @@
    */
   observation?: Observation[];
   /**
-   * A reference to - The observing mode
+   * a configuration can be used to observe with.
    */
   mode?: ObservingMode;
 };
@@ -635,6 +652,9 @@
  * An organisation that can perform astronomical observations
  */
 export type Observatory = {
+  "@type"?: string;
+  "_id"?: number;
+
   xmlId?: string;
   /**
    * The name of the organization
@@ -642,7 +662,7 @@
   name?: string;
   address?: string;
   /**
-   * the registry identifier for the organization
+   * an identifier that can be used as a key to look up in an IVOA registry - see https://www.ivoa.net/documents/IVOAIdentifiers/
    */
   ivoid?: Ivoid;
   /**
@@ -680,6 +700,7 @@
  * a configuration can be used to observe with.
  */
 export type ObservingMode = {
+  _id?: number;
   /**
    * human readable name for the mode
    */
@@ -689,19 +710,19 @@
    */
   description?: string;
   /**
-   * A reference to -
+   * base type of a telescope or array of telescopes
    */
   telescope?: ObservingPlatform;
   /**
-   * A reference to -
+   * An instrument that can be attached to a telescope - e.g. CCD, Radio Receiver
    */
   instrument?: Instrument;
   /**
-   * The filter or frequency response that
+   * Available filters /bandpasses for intruments
    */
   filter?: Filter;
   /**
-   * A reference to -
+   * a processing backend /pipeline- e.g. correlator
    */
   backend?: Backend;
   xmlId?: string;
@@ -711,6 +732,7 @@
  * base type of a telescope or array of telescopes
  */
 export type ObservingPlatform = {
+  _id?: number
   xmlId?: string;
 };
 
@@ -718,6 +740,7 @@
  * a complete proposal
  */
 export type ObservingProposal = {
+  _id?: number;
   xmlId?: string;
   /**
    * the proposal title
@@ -728,11 +751,11 @@
    */
   summary?: string;
   /**
-   * scientific justification
+   * The justification for the proposal. Note that the justification is for reading by humans and is not parsed to extract things like source lists - that sort of information must be entered in the correct place in the model.
    */
   scientificJustification?: Justification;
   /**
-   * technical justification
+   * The justification for the proposal. Note that the justification is for reading by humans and is not parsed to extract things like source lists - that sort of information must be entered in the correct place in the model.
    */
   technicalJustification?: Justification;
   /**
@@ -770,6 +793,8 @@
  * An institution that is a collection of people
  */
 export type Organization = {
+  "@type"?: string; // proposal:Organization
+  _id?: number;
   /**
    * The name of the organization
    */
@@ -821,6 +846,7 @@
    */
   orcidId?: StringIdentifier;
   xmlId?: string;
+  _id?: number;
 };
 
 /**
@@ -865,6 +891,7 @@
  * Defines collection of resources and proposals for a particular observing season
  */
 export type ProposalCycle = {
+  _id?: number
   /**
    * A reference to -
    */
@@ -883,20 +910,32 @@
   instructions?: string;
   /**
    * the date by which observing proposals must be submitted to be considered for this cycle
+   *
+   * @format date
+   * @example "2022-03-10T00:00:00.000Z"
    */
   submissionDeadline?: Date;
   /**
    * the start date of the observing cycle
+   *
+   * @format date
+   * @example "2022-03-10T00:00:00.000Z"
    */
   observationSessionStart?: Date;
   /**
    * the end date of the observing cycle
+   *
+   * @format date
+   * @example "2022-03-10T00:00:00.000Z"
    */
   observationSessionEnd?: Date;
   /**
    * the possible observing modes offered in the cycle
    */
   observingModes?: ObservingMode[];
+  /**
+   * The list of resources that are available
+   */
   availableResources?: AvailableResources;
   /**
    * The possible grades that can be allocated for this cycle
@@ -931,8 +970,9 @@
  * A review of a proposal
  */
 export type ProposalReview = {
+  _id?: number;
   /**
-   * A reference to - the reviewer making this review
+   * assigned to review the proposal
    */
   reviewer?: Reviewer;
   /**
@@ -951,6 +991,9 @@
   technicalFeasibility?: boolean;
   /**
    * the date that the review is submitted
+   *
+   * @format date
+   * @example "2022-03-10T00:00:00.000Z"
    */
   reviewDate?: Date;
 };
@@ -982,7 +1025,7 @@
  */
 export type Quantity = {
   /**
-   * The unit of this quantity.
+   * Must conform to definition of unit in VOUnit spec.
    */
   unit?: Unit;
 };
@@ -991,6 +1034,7 @@
  * A real value with a unit.
  */
 export type RealQuantity = {
+  "@type"?: string; // ivoa:RealQuantity
   /**
    * The unit of this quantity.
    */
@@ -1008,7 +1052,7 @@
  */
 export type RelatedProposal = {
   /**
-   * A reference to - the related proposal
+   * base proposal
    */
   proposal?: AbstractProposal;
 };
@@ -1017,8 +1061,9 @@
  * A resource that will be consumed by allocating an observation from a proposal
  */
 export type Resource = {
+  _id?: number;
   /**
-   * A reference to - the type of the resource
+   * a type of resource
    */
   type?: ResourceType;
   /**
@@ -1034,11 +1079,11 @@
  */
 export type ResourceBlock = {
   /**
-   * A reference to -
+   * a configuration can be used to observe with.
    */
   mode?: ObservingMode;
   /**
-   * the amount of resource
+   * A resource that will be consumed by allocating an observation from a proposal
    */
   resource?: Resource;
 };
@@ -1047,6 +1092,7 @@
  * a type of resource
  */
 export type ResourceType = {
+  _id?: number;
   /**
    * the name of the resource type
    */
@@ -1059,8 +1105,9 @@
  * assigned to review the proposal
  */
 export type Reviewer = {
+  _id?: number;
   /**
-   * A reference to - the person taking on the reviewer role
+   * person connected with the proposal
    */
   person?: Person;
   xmlId?: string;
@@ -1087,6 +1134,11 @@
  * A spectral window for science use.
  */
 export type ScienceSpectralWindow = {
+  /**
+   * Science oriented definition of a spectral window.
+   */
+  _id?: number;
+
   spectralWindowSetup?: SpectralWindowSetup;
   expectedSpectralLine?: ExpectedSpectralLine[];
 };
@@ -1169,7 +1221,7 @@
    */
   _id?: number;
   /**
-   * the person here
+   * person connected with the proposal
    */
   person?: Person;
   uid?: string;
@@ -1191,6 +1243,8 @@
  * an instance of a proposal that has been submitted
  */
 export type SubmittedProposal = {
+  _id?: number;
+
   xmlId?: string;
   /**
    * the proposal title
@@ -1201,11 +1255,11 @@
    */
   summary?: string;
   /**
-   * scientific justification
+   * The justification for the proposal. Note that the justification is for reading by humans and is not parsed to extract things like source lists - that sort of information must be entered in the correct place in the model.
    */
   scientificJustification?: Justification;
   /**
-   * technical justification
+   * The justification for the proposal. Note that the justification is for reading by humans and is not parsed to extract things like source lists - that sort of information must be entered in the correct place in the model.
    */
   technicalJustification?: Justification;
   /**
@@ -1244,6 +1298,9 @@
   config?: ObservationConfiguration[];
   /**
    * the date that the proposal was submitted
+   *
+   * @format date
+   * @example "2022-03-10T00:00:00.000Z"
    */
   submissionDate?: Date;
   /**
@@ -1252,6 +1309,9 @@
   successful?: boolean;
   /**
    * the date when all the proposals are due
+   *
+   * @format date
+   * @example "2022-03-10T00:00:00.000Z"
    */
   reviewsCompleteDate?: Date;
   /**
@@ -1313,17 +1373,20 @@
  * A target source
  */
 export type Target = {
+  "@type"?: string;
   /**
    * A common name for the source
    */
   sourceName?: string;
   xmlId?: string;
+  _id?: number;
 };
 
 /**
  * the field points to the associated target
  */
 export type TargetField = {
+  "@type"?: string;
   xmlId?: string;
   name?: string;
 };
@@ -1332,6 +1395,7 @@
  * an observation of the scientific target
  */
 export type TargetObservation = {
+  "@type"?: string;
   xmlId?: string;
   /**
    * A reference to - The actual target of the observation
@@ -1355,12 +1419,18 @@
  * collects together the technical goals of the proposal
  */
 export type TechnicalGoal = {
+  /**
+   * The parameters required for observation to be useful for the science goal
+   */
   performance?: PerformanceParameters;
   spectrum?: ScienceSpectralWindow[];
+  _id?: number;
   xmlId?: string;
 };
 
 export type Telescope = {
+  _id?: number
+  "@type"?: string;
   xmlId?: string;
   /**
    * telescope name
@@ -1380,6 +1450,7 @@
  * a set of telescopes that are operated together for an observation
  */
 export type TelescopeArray = {
+  "@type"?: string;
   xmlId?: string;
   /**
    * the array name
@@ -1392,9 +1463,6 @@
 };
 
 export type TelescopeArrayMember = {
-  /**
-   * A reference to -
-   */
   telescope?: Telescope;
 };
 
@@ -1415,9 +1483,21 @@
  * particular time range
  */
 export type TimingWindow = {
+  "@type"?: string; //proposal:TimingWindow
+
+  _id?: number;
+
   note?: string;
   isAvoidConstraint?: boolean;
+  /**
+   * @format date
+   * @example "2022-03-10T00:00:00.000Z"
+   */
   startTime?: Date;
+  /**
+   * @format date
+   * @example "2022-03-10T00:00:00.000Z"
+   */
   endTime?: Date;
 };
 
